pkg/cmd/experimental/syncgroups/grouplister.go

```diff
@@ -0,0 +1,123 @@
+package syncgroups
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+	syncutil "github.com/openshift/origin/pkg/cmd/experimental/syncgroups/util"
+	ouserapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// NewAllOpenShiftGroupLister returns a new AllLocalGroupLister
+func NewAllOpenShiftGroupLister(ldapURL string, groupClient osclient.GroupInterface) interfaces.LDAPGroupLister {
+	return &AllLocalGroupLister{
+		client:  groupClient,
+		ldapURL: ldapURL,
+	}
+}
+
+// AllLocalGroupLister lists unique identifiers for LDAP lookup of all local OpenShift Groups that
+// have been marked with an LDAP URL annotation as a result of a previous sync.
+type AllLocalGroupLister struct {
```

Comment:
I don't think it was. If you don't change your config, you will be deterministically mapping these groups to their names anyway. The  dynamic `GroupNameMapper` uses a `GroupGetter` (which you commented on looking bad earlier). This `GroupGetter` necessarily uses a `GroupLister` to gain information about groups in the first place, which brings us back to this object. It does in the end guarantee that you sync the same group to the same place twice.